middlewares/team.middleware.js

```diff
@@ -165,6 +165,35 @@ async function findById(req, res, next) {
     return next();
 }
 
+/**
+ * @async
+ * @function deleteHackerFromTeam
+ * @param {{user: {id: ObjectId}} req 
+ * @param {*} res 
+ * @return {JSON} Success or error status
+ * @description Removes the hacker from the team under teamId. If hacker is not part of a team, it does nothing.
+ */
+async function deleteHackerFromTeam(req, res, next) {
+    const hacker = await Services.Hacker.findByAccountId(req.user.id);
+
+    if (!hacker) {
+        return next({
+            status: 404,
+            message: Constants.Error.HACKER_404_MESSAGE,
+            data: {
+                id: req.user.id
+            }
+        });
+    }
+    const oldTeamId = hacker.teamId;
+    if (oldTeamId) {
+        const result1 = await Services.Team.removeMember(oldTeamId, hacker._id);
+        const result2 = await Services.Team.removeTeamIfEmpty(oldTeamId);
+        console.log(result1, result2);
+    }
+    next();
```

Comment:
Should we put in some kind of error if oldTeamId doesn't exist? What do you think should be the output if the hacker was never part of a team?

Reply:
Since the resulting database state is the same, returning success should be fine.

Reply:
Sounds good